tests/api/test_connection.py

```diff
@@ -260,6 +280,30 @@ def function(*args, **kwargs):
         # A successful cancellation does not raise an exception
         connection.cancel_job("123")
 
+    def test_cancel_pending_job(self, connection, monkeypatch):
+        """Tests a job cancellation first is pending and then is successful by
+        using a mock server."""
+        # A custom `mock_return` that checks for expected arguments
+        def _mock_return(return_value):
+            def function(*args, **kwargs):
+                assert kwargs.get("json") == {"status": "cancelled"}
+                return return_value
+
+            return function
+
+        server = MockCancelerServer()
+        monkeypatch.setattr(requests, "patch", server.patch)
+
+        # A successful cancellation does not raise an exception
+        connection.cancel_job("123")
+
+        # Job cancellation is first pending
+        server.request_dict["123"] == JobStatus.CANCEL_PENDING.value
+        sleep(0.2)
+
+        # Job cancellation is successful
+        server.request_dict["123"] == JobStatus.CANCELLED.value
+
```

Comment:
Also, I wonder if we're really testing anything meaningful in this particular case: since we manually set the value, then check the value directly, it feels like we are validating the behaviour of `time.sleep` more than anything else :smiley: 

I think the transition from `cancel_pending` to `cancelled` might be better tested at the integration level between SF and the platform API, which would help to simplify things here a bit. But let me know if I'm missing something important.